### ISSUE TITLE
I hope that QGIS will add a new interface style, like the Ribbon style of Office and Arcgis pro

### ISSUE THREAD

**Issue by tyuhao** [qgis/QGIS#52292]
### Feature description

I think the ribon style is more efficient and beautiful

**Comment by bkepl** [comment]
In my opinion, the ribbon adds unnecessary clicks to get to anything. I agree, it's "more beautiful," but when using ArcGIS Pro, I find it takes too many clicks to do anything, and I don't always agree with the groupings Esri has made. Even more annoying is (for example) the fact that, in the case of labeling, not all features can be accessed in the panels; some features can only be accessed through the ribbon. 

So if QGIS eventually goes this route, I would hope features accessible in the panels wouldn't be split up in like manner. 

The fact is, I do most of my work interfacing with the panels and relatively little through the menus or the toolbars. 

There *are* two things I'd like to see: 

1) The ability to move toolbars to a separate dockable panel, only keeping the ones I use most frequently front-and-center (I know I can selectively hide or display toolbars, which I may do more of). 
2) More consistent placement of menu items and icons created by plugins, which I know is the choice of the plugin author, so impossible to control.

**Comment by andreimirt** [comment]
@bkepl If a ribbon interface will exist, it should have different _work moods_, and even those should be entirely customizable. By _work moods_ I mean that groupings of different tools/options should be based on the primary kind of work that is being done. QGIS is a very complex software and its users use it in many different ways. Some use it primarily for drafting, others do data analysis, while others just visualization. Each of these types of users use some tools more than others and have some specific workflows. Each popular type of user should have its own work mood, and further, each subtype of user can customize its own interface as they wish. Sometimes the same user does drafting on Mondays and processing on Wednesdays, so they may switch to a different _work mood_. Autodesk implements in its software such a design principle. From my point of view they are doing Ribbon style UI the best, so perhaps that's something to aspire to.

This _work mood_ principle can probably be applied to any kind of design, not just to Ribbon style UI. Even the current toolbar based design can probably benefit from this principle. Actually, this is almost already implemented via profiles. One could just set up different profiles with different active toolbars and a different layout, depending on what kind of work is necessary. There's 2 minor issues, 1. profiles are not only about UI, but every possible QGIS setting, including installed plugins (but I guess this could also be an advantage), and 2. there are no predefined profiles for popular workflows/typical users.

With the rise of webapps in the past decade, in my opinion, better UI designs have emerged than Ribbon UI.  I'm not sure if Ribbon UI would not feel dated if implemented. So it might not be worth it. Probably the better direction would be some webapp like design style, but I still find it difficult to justify the effort of switching.

**Comment by paul-dorsetcouncil** [comment]
I'm not sure I want to see a ribbon as such, although perhaps it is needed to keep pace with other tools, but I do think the UI needs looking at. We often get the dialogues slowly moving down the screen on Windows because it basically has the look and feel of an old Windows form application. We also find it does glitch a fair amount and I think that is in part due to the UI. Hoping the upgrade of Qt will be enough but it would be nice to hear more about the alternative options at some point; is Qt the only option? Have I missed a talk on this that would be worth a watch? 

It is definitely the most common comment I get from QGIS users. The UI feels old fashioned. That is not to criticize the amazing developers! QGIS has opened up GIS tools to so many and I know many close to the project probably actually like that it still has menus; but just to reflect the obvious really. I mean it is accepted in that the upgrade to Qt6 is underway...but I'm just hoping it will allow enough modernisation and improved reliability to allow QGIS to carry on for many more years to come.

**Comment by nickrsan** [comment]
Seeing this is still open, adding my two cents. LibreOffice offers an optional ribbon-style UI, which I think is a great way to go for FOSS software. Don't force the change for the many people who find the existing flow more efficient and allow the switch for those who find Ribbons more efficient.

I personally have found ribbons much more efficient, and much easier to *teach* to people than icon/menu-based interfaces, so I look to see if complex software has it, but I see QGIS doesn't. I also see there is a plugin that appears abandoned that would add this, and maybe it makes the most sense as a plugin supported by those who want it, but if there was a way to optionally include it in core, I think that would be fantastic.

**Comment by pfolk** [comment]
I agree that a ribbon-style interface would be helpful.  My take is that QGIS has SO MANY great tools and toolbars, that other than the key features I use daily it's hard to keep track of things.  I end up turning off lots of toolbars just to get the UI to be usable and yet I still have at least 2, and often 3, layers of toolbars, and some of the toolbars only partially shown.  This is barely manageable for me, and I use the tool quite a lot and have zero problems with interface overwhelm.  For any "normal user," the number of buttons to deal with would be such that they basically would never use the vast majority of features.

The ribbon interface gives hierarchy to the tools.  Keyboard shortcuts + Ribbon, imo, provides the best of both worlds:
* Discoverability due to more structured, less cluttered feature layout in the ribbon
* Instant access to frequently used features
* Interface parity with ArcGIS

Presumably, because it's FOSS, anyone who prefers clutter can presumably opt out of the ribbon end end up back in their own personal toolbar hell =)

I hope someone picks up this ball and runs with it.  Tag me if you would be interested in doing so; I would be happy to sponsor the development.

**Comment by anitagraser** [comment]
~Some experimental work is here: https://github.com/wonder-sk/QGIS/commit/f2801fefadee0612f015d183aaca69fae9cf8fed~

Updated: scratch that :)

**Comment by anitagraser** [comment]
There is also an experimental plugin: https://plugins.qgis.org/plugins/ribbon_toolbar/#plugin-details 

<img width="1403" height="445" alt="Image" src="https://github.com/user-attachments/assets/3e9f4f5b-5a18-455d-9311-520b018673f2" />

**Comment by wonder-sk** [comment]
> Some experimental work is here: https://github.com/wonder-sk/QGIS/commit/f2801fefadee0612f015d183aaca69fae9cf8fed

Note: that was built just as a April's fools joke 😁

**Comment by nirvn** [comment]
@wonder-sk , you're not into relying on a 24-month-old Chinese repo as the basis of the ribbon UI? 😉

**Comment by anitagraser** [comment]
> Note: that was built just as a April's fools joke 😁

Got me :) Let me clean that up then

**Comment by froehlicher** [comment]
I also would like to see a ribbon style GUI native to QGIS.

This Plugin here is pretty much what it should look and feel like: https://github.com/M-Rick/QGIS-Ribbon.git
Its customizable, you can add toolbars, tools and custom tabs to it and and the same time choose to show or hide the menus up top. 
The only thing that is missing, is the ability to have drop downs or fly outs of tools that do similar things.

I can highly recommend it.

<img width="1280" height="231" alt="Image" src="https://github.com/user-attachments/assets/4a4b5d9f-e8f2-4a4a-ab6f-c1cf904d593c" />

**Comment by anitagraser** [comment]
Would love your feedback on this PoC: https://github.com/anitagraser/qgis-tabbed-ui

It's inspired by LibreOffice's ribbon theme option (as mentioned by @nickrsan) and targets new users / teaching situations. Therefore, commonly used buttons are larger (easy to find) and I tried to design logical groups.

I've mostly focused on the "Project" and "Home" ribbons so far. The others still need love for sure. 

I haven't implemented any way yet to customize the ribbons. Of course, this could be added at a later stage. In any case, the user can, of course, switch back to good old toolbars at any time. 

Disclaimer: The code is NOT intended to be production-ready. Consider this a click-dummy that should facilitate discussion. 

<img width="1572" height="956" alt="Image" src="https://github.com/user-attachments/assets/c852f68a-8282-4abb-92ba-55e4a2a6eca0" />

The dark theme is "Studio Premium" from [QGIS Studio Themes](https://github.com/GallPeters/qgis-studio-themes).

**Comment by eithanwes** [comment]
Hey, that's [my plugin](https://github.com/eithanwes/ribbon_toolbar)! Really cool to see people looking at it!

I'm glad there's an actual discussion happening around this topic, and I'm totally for it.

I started it as a basic POC and set it aside when I couldn't put much more time into refining the idea. @anitagraser 's version is a giant leap in user experience from what I tried.

I'd love to help move this forward however I can, if you want you can PR the plugin and I'll publish a new version.

**Comment by anitagraser** [comment]
Glad to see that you're still around, @eithanwes! Your plugin was a great starting point for this work and I enriched it with design ideas from LibreOffice.

I've changed the module name so it could be published separately but I'm happy to change it back if you want to publish it. Let me know what you prefer.

**Comment by eithanwes** [comment]
I'll be happy to continue publishing it and updating it. 
This gave me the push to work on a customisation feature

<img width="955" height="554" alt="Image" src="https://github.com/user-attachments/assets/1ec53a2c-08d1-424c-9361-9232bb1f082f" />
